Drop-In/src/androidTest/java/com/braintreepayments/api/SupportedPaymentMethodsFragmentUITest.kt

```diff
@@ -0,0 +1,259 @@
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        val cardNonce = CardNonce.fromJSON(JSONObject(Fixtures.PAYMENT_METHODS_VISA_CREDIT_CARD))
+        val vaultedPaymentMethods = listOf(cardNonce)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+            fragment.dropInViewModel.setVaultedPaymentMethods(vaultedPaymentMethods)
+        }
+
+        onView(withId(R.id.bt_vault_edit_button)).check(matches(isDisplayed()))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenVaultManagerIsDisabled_hidesVaultEditButton() {
+        val dropInRequest = DropInRequest()
+        dropInRequest.isVaultManagerEnabled = false
+        val bundle = bundleOf("EXTRA_DROP_IN_REQUEST" to dropInRequest)
+
+        val scenario = FragmentScenario.launchInContainer(SupportedPaymentMethodsFragment::class.java, bundle)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+        }
+
+        onView(withId(R.id.bt_vault_edit_button)).check(matches(not(isDisplayed())))
+    }
+
+    @Test
+    fun whenStateIsRESUMED_whenCreditOrDebitCardSelected_sendsPaymentMethodSelectedEvent() {
+        val dropInRequest = DropInRequest()
+        dropInRequest.isVaultManagerEnabled = false
+        val bundle = bundleOf("EXTRA_DROP_IN_REQUEST" to dropInRequest)
+
+        val scenario = FragmentScenario.launchInContainer(SupportedPaymentMethodsFragment::class.java, bundle)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedPaymentMethods(supportedPaymentMethods)
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withText("Credit or Debit Card")).perform(click())
+        scenario.onFragment { fragment ->
+            val activity = fragment.requireActivity()
+            val fragmentManager = fragment.parentFragmentManager
+            fragmentManager.setFragmentResultListener(DropInEvent.REQUEST_KEY, activity) { _, result ->
+                val event = DropInEvent.fromBundle(result)
+                assertEquals(DropInEventType.SUPPORTED_PAYMENT_METHOD_SELECTED, event.type)
+
+                val paymentMethodType =
+                    event.getDropInPaymentMethodType(DropInEventProperty.SUPPORTED_PAYMENT_METHOD)
+                assertEquals(DropInPaymentMethodType.UNKNOWN, paymentMethodType)
+                countDownLatch.countDown()
+            }
+        }
+
+        countDownLatch.await()
+    }
+
+    @Test(timeout = 5000)
+    fun whenStateIsRESUMED_whenVaultManagerEnabledAndVaultedCardExists_sendsAnalyticEvent() {
+        // TODO: capture all analytics events within a time interval and assert that the target analytics event is emitted
```

Comment:
Do we want to keep this commented out?  If so, could we add why so we have a reference as to what we can do to uncomment it?